master/internal/taskconfigpolicy/task_config_policy.go

```diff
@@ -0,0 +1,41 @@
+package taskconfigpolicy
+
+import (
+	"github.com/determined-ai/determined/master/pkg/model"
+	"github.com/determined-ai/determined/master/pkg/schemas/expconf"
+)
+
+// ResourceConstraints are non-overridable resource constraints.
+// Submitted workloads that request resource quanities exceeding defined resource constraints in a
+// given scope are rejected.
+type ResourceConstraints struct {
+	MaxSlots    *int `json:"max_slots"`
+	MaxPriority *int `json:"max_priority"`
+}
+
+// Constraints are non-overridable workload constraints.
+// Submitted workloads whose config's respective field(s) exceed defined constraints within a given
+// scope are rejected.
+type Constraints struct {
+	ResourceConstraints ResourceConstraints `json:"resources"`
+}
+
+// ExperimentTaskConfigPolicy is the invariant config and constraints for an experiment.
+// Submitted experiments whose config fields vary from the respective InvariantConfig fields set
+// within a given scope are silently overriden.
+// Submitted experiments whose constraint fields vary from the respective Constraint fields set
+// within a given scope are rejected.
+type ExperimentTaskConfigPolicy struct {
```

Comment:
Do you like `ExperimentTaskConfigPolicy` or `ExperimentConfigPolicies` better? I'm ok with either; I just wonder if `experiment` and `task` is a little repetitive?

Reply:
I prefer `ExperimentConfigPolicy` and `NTSCConfigPolicy`, changed!